Assignment7.txt

```diff
@@ -0,0 +1,22 @@
+//Anya Lauria 
+
+1. ++*p increments the variable that the pointer is pointing to first, and then evaluates it, similar to prefix 
+   incrementing in regular variables. *p++ evaluates the variable that the  pointer is pointing to first, then 
+   advances it. Not sure about *++p 
+2. The left to right order and right to left order depends on the kind of operation. For example, post fix operations 
+   always take precedence. The order can be manipulated using parantheses. 
+3. You can use them in function arguments, and not copy large chunks of memory, so the use of pointers is efficient. 
+   Pointers are also used to return arrays- there is no other way. Instead of using a character array, we can use 
+   pointers to work with strings instead, which is much easier. They also allow us to dynamic memory allocation, and
+   resize data structure whenever needed. 
+4.  4.1 char * 
+    4.2 invalid 
+    4.3 0==0 returns 1, so 1 
+    4.3 Int 
+    4.4 Int *- this is creating a pointer for a[0]
+    4.5 Int 
+    4.6 Int **- creates a pointer for pointer p 
+    4.8 I didn't understand this 
+    4.9 Invalid, you can't create a pointer of a function 
```

Comment:
Functions in C are actually just pointers to a spot in the program where some code exists. Just like you've been creating pointers to structs, strings, and arrays, you can point a pointer at a function too. The main use for this is to pass "callbacks" to other functions, or to simulate classes and objects. In this exercise we'll do some callbacks, and in the next one we'll make a simple object system.

The format of a function pointer goes like this:

```int (*POINTER_NAME)(int a, int b)```

So the answer is actually ```int(*)(int, char**). All you needed to do was evaluate the data types of the &main. Since & returns the memory address, main's data type would change from int to int*. Since argc and argv are just the parameters, they would remain the same data types. So the final answer is int(*)(int, char**).```